app/models/concerns/language_and_schedule_references.rb

```diff
@@ -5,6 +5,20 @@ module LanguageAndScheduleReferences
     has_many :language_skills, as: :languageable, dependent: :destroy
     accepts_nested_attributes_for :language_skills, allow_destroy: true
 
+    def first_language
+      return [] if language_skills.blank? || language_skills.native.blank?
+      return language_skills.native.german if language_skills.native.german.present?
+      return language_skills.native.limit(1) if language_skills.native.size.positive?
+    end
```

Comment:
Alternative, more readable IMO:

1. leave the guard as is, as it checks the precondition
2. The rest could be done as a simple OR statement

```ruby
return if language_skills.blank? || language_skills.native.blank?

language_skills.native.german.first || language_skills.native.first
```

You could also extract a new scope that returns the first DE or first otherwise.